common/utils/src/main/resources/error/error-classes.json

```diff
@@ -846,6 +846,11 @@
       "Exceeds char/varchar type length limitation: <limit>."
     ]
   },
+  "EXCEED_RETRY_JVM" : {
```

Comment:
If this needs to use a different name than python, I'd vote for just making it "RETRIES_EXCEEDED", because the "_JVM" in the name is confusing for a user facing name. @itholic @HyukjinKwon...

Reply:
`RETRIES_EXCEEDED` seems fine to me.

Btw, I just realized that the existing error message was `The number of retries ($currentRetryNum) must not exceed the maximum number of retires (${retryPolicy.maxRetries}).`, but the error message here `Retries exceeded but no exception caught.` that just mimic the Python side?? I think maybe the current error message doesn't explain the situation properly??

When adding an error class, generally we keep the existing error message or improve it if necessary. In the case of the error class that exists in Python, the existing error message is kept as it is, so the situation and details on the JVM side may be different.

<img width="535" alt="Screenshot 2023-08-11 at 9 51 02 AM" src="https://github.com/apache/spark/assets/44108233/46278f4f-8b58-42c5-bd15-167f7b8847f6">

So, at least I'd recommended to keep the existing error message something like:
```json
"RETRIES_EXCEEDED" : {
  "message" : [
    "The number of retries <currentRetryNum> must not exceed the maximum number of retires (<retryPolicy.maxRetries>)."
  ]
},
```

Reply:
Yeah, that message in python is strange, but since this PR tried to do the same, I didn't nit on that.
But as @cdkrot mentions in https://github.com/apache/spark/pull/42399#discussion_r1290807582, python also hides the underlying exception